CMakeLists.txt

```diff
@@ -48,6 +48,8 @@ if(BUILD_TUTORIALS)
 #    add_subdirectory(examples/rabbitmq_tutorials)
 endif()
 
+option(BUILD_PURE "build in pure mode" OFF)
```

Comment:
I've seen projects following either way, so maybe it is a bit controversial here.  Some may feel it like `--with-tutorials` or `--without-tcp`, and others still are afraid of name conflicts.

Name conflicts come into play when this project is used by `add_subdirectory` from another project.  Looking at that set to parent scope, seems we are to support this use case?

Personally I prefer to prefix it, but here I'm just following existing code --- several lines above, `BUILD_TUTORIALS`.